server/src/main/java/org/opensearch/cluster/routing/remote/RemoteRoutingTableService.java

```diff
@@ -0,0 +1,114 @@
+import org.opensearch.cluster.ClusterState;
+import org.opensearch.cluster.routing.RoutingTable;
+import org.opensearch.common.settings.ClusterSettings;
+import org.opensearch.common.settings.Setting;
+import org.opensearch.common.settings.Settings;
+import org.opensearch.common.util.io.IOUtils;
+import org.opensearch.gateway.remote.ClusterMetadataManifest;
+import org.opensearch.node.Node;
+import org.opensearch.node.remotestore.RemoteStoreNodeAttribute;
+import org.opensearch.repositories.RepositoriesService;
+import org.opensearch.repositories.Repository;
+import org.opensearch.repositories.blobstore.BlobStoreRepository;
+
+import java.io.Closeable;
+import java.io.IOException;
+import java.util.List;
+import java.util.function.Supplier;
+
+import static org.opensearch.node.remotestore.RemoteStoreNodeAttribute.isRemoteRoutingTableEnabled;
+
+/**
+ * A Service which provides APIs to upload and download routing table from remote store.
+ *
+ * @opensearch.internal
+ */
+public class RemoteRoutingTableService implements Closeable {
+
+    /**
+     * Cluster setting to specify if routing table should be published to remote store
+     */
+    public static final Setting<Boolean> REMOTE_ROUTING_TABLE_ENABLED_SETTING = Setting.boolSetting(
+        "cluster.remote_store.routing_table.enabled",
+        false,
+        Setting.Property.NodeScope,
+        Setting.Property.Final
+    );
+    private static final Logger logger = LogManager.getLogger(RemoteRoutingTableService.class);
+    private final Settings settings;
+    private final Supplier<RepositoriesService> repositoriesService;
+    private final ClusterSettings clusterSettings;
+    private BlobStoreRepository blobStoreRepository;
+
+    public RemoteRoutingTableService(
+        Supplier<RepositoriesService> repositoriesService,
+        Settings settings,
+        ClusterSettings clusterSettings
+    ) {
+        assert isRemoteRoutingTableEnabled(settings) : "Remote routing table is not enabled";
+        this.repositoriesService = repositoriesService;
+        this.settings = settings;
+        this.clusterSettings = clusterSettings;
+    }
+
+    public List<ClusterMetadataManifest.UploadedIndexMetadata> writeFullRoutingTable(
+        ClusterState clusterState,
+        String previousClusterUUID
+    ) {
+        return null;
+    }
+
+    public List<ClusterMetadataManifest.UploadedIndexMetadata> writeIncrementalMetadata(
+        ClusterState previousClusterState,
+        ClusterState clusterState,
+        ClusterMetadataManifest previousManifest
+    ) {
+        return null;
+    }
+
+    public RoutingTable getLatestRoutingTable(String clusterName, String clusterUUID) {
+        return null;
+    }
+
+    public RoutingTable getIncrementalRoutingTable(
+        ClusterState previousClusterState,
+        ClusterMetadataManifest previousManifest,
+        String clusterName,
+        String clusterUUID
+    ) {
+        return null;
+    }
```

Comment:
if we need to push this, we might hv to remove unimplemented methods

Reply:
Ack, removed for now. Will add back in the next few PRs when we add impl of these.